File: Eulerjs/Euler3.js
var num = 600851475143;
var l, m, i, n;

function isprime(n){
	if( n % 2 == 0){ return false;}
	for(i = 3; i < Math.pow(n, .5); i++){
		if( (n%i) == 0 ){ return false; }
	}
	return true;
}

for( l = 0; l < Math.pow(num, .5)/2; l++){
	if( (num % l == 0) && isprime(l)){
		m = l;
	}

}

console.log(m);

File: Eulergo/Euler3.go
package main

import "fmt"
import "math"

func isprime(a int) bool {
	if a < 1 || a % 2 == 0 {
		return false
	}
	if a == 2 {
		return true
	}
	f := math.Sqrt(float64(a))
	y := int(f)
	for i := 3; i <= y + 1; i += 2 {
		if a % i == 0 {
			return false
		}
	}
	return true
}

func main(){
	num := 600851475143
	stuff := 0
	for i := 3; i <= num; i += 2 {
		if num % i == 0 && isprime(i) {
			stuff = i
		}
	}
	fmt.Println(stuff)
}
